pkg/storageops/azure/azure.go

```diff
@@ -0,0 +1,808 @@
+	snapshotsClient   *compute.SnapshotsClient
+}
+
+func (a *azureOps) Name() string {
+	return Name
+}
+
+func (a *azureOps) InstanceID() string {
+	return a.instance
+}
+
+func NewEnvClient() (storageops.Ops, error) {
+	instance, err := storageops.GetEnvValueStrict(EnvInstanceName)
+	if err != nil {
+		return nil, err
+	}
+
+	subscriptionID, err := storageops.GetEnvValueStrict(EnvSubscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceGroupName, err := storageops.GetEnvValueStrict(EnvResourceGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewAzureClient(instance, subscriptionID, resourceGroupName)
+}
+
+func NewAzureClient(
+	instance, subscriptionID, resourceGroupName string,
+) (storageops.Ops, error) {
+	envName := os.Getenv(EnvAzureEnvironment)
+	if len(envName) == 0 {
+		envName = defaultEnvironment
+	}
+	env, err := azure.EnvironmentFromName(envName)
+	if err != nil {
+		return nil, fmt.Errorf("invalid cloud name '%s' specified: %v", envName, err)
+	}
+
+	tenantID, err := storageops.GetEnvValueStrict(EnvTenantID)
+	if err != nil {
+		return nil, err
+	}
+
+	clientID, err := storageops.GetEnvValueStrict(EnvClientID)
+	if err != nil {
+		return nil, err
+	}
+
+	clientSecret, err := storageops.GetEnvValueStrict(EnvClientSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	oauthConfig, err := adal.NewOAuthConfig(env.ActiveDirectoryEndpoint, tenantID)
+	if err != nil {
+		return nil, err
+	}
+
+	token, err := adal.NewServicePrincipalToken(*oauthConfig, clientID,
+		clientSecret, env.ResourceManagerEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	authorizer := autorest.NewBearerAuthorizer(token)
+
+	disksClient := compute.NewDisksClient(subscriptionID)
+	disksClient.Authorizer = authorizer
+	disksClient.PollingDelay = clientPollingDelay
+	disksClient.RetryAttempts = clientRetryAttempts
+	disksClient.AddToUserAgent(userAgentExtension)
+
+	vmsClient := compute.NewVirtualMachinesClient(subscriptionID)
+	vmsClient.Authorizer = authorizer
+	vmsClient.PollingDelay = clientPollingDelay
+	vmsClient.RetryAttempts = clientRetryAttempts
+	vmsClient.AddToUserAgent(userAgentExtension)
+
+	snapshotsClient := compute.NewSnapshotsClient(subscriptionID)
+	snapshotsClient.Authorizer = authorizer
+	snapshotsClient.PollingDelay = clientPollingDelay
+	snapshotsClient.RetryAttempts = clientRetryAttempts
+	snapshotsClient.AddToUserAgent(userAgentExtension)
+
+	return &azureOps{
+		instance:          instance,
+		resourceGroupName: resourceGroupName,
+		disksClient:       &disksClient,
+		vmsClient:         &vmsClient,
+		snapshotsClient:   &snapshotsClient,
+	}, nil
+}
+
+func (a *azureOps) Create(
+	template interface{},
+	labels map[string]string,
+) (interface{}, error) {
+	d, ok := template.(*compute.Disk)
+	if !ok {
+		return nil, storageops.NewStorageError(
+			storageops.ErrVolInval,
+			"Invalid volume template given",
+			a.instance,
+		)
+	}
+
+	ctx := context.Background()
+	future, err := a.disksClient.CreateOrUpdate(
```

Comment:
Is the behavior for Create() different from `aws` because here it will also update the volume if it already exists?. We should check if it already exists and return an error if that is the case.